incorrect self update command provided

### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

_No response_

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Link to the code that reproduces this issue or a replay of the bug

n/a

### Reproduction steps

Run a command that checks the currently installed version,
e.g.
`pnpm add -g typescript ts-node`

```matthew@MacBookPro ~/code> uname -a && brew --version
Darwin MacBookPro.local 24.0.0 Darwin Kernel Version 24.0.0: Mon Aug 12 20:54:26 PDT 2024; root:xnu-11215.1.10~2/RELEASE_ARM64_T8112 arm64
Homebrew 4.3.23
```

Node and Corepack installed via homebrew and latest versions installed:
matthew@MacBookPro ~/code> node -v && corepack --version
v22.9.0
0.29.4

 

### Describe the Bug

Instructions provided:
```
   ╭──────────────────────────────────────────────────────────────────╮
   │                                                                  │
   │                Update available! 9.9.0 → 9.10.0.                 │
   │   Changelog: https://github.com/pnpm/pnpm/releases/tag/v9.10.0   │
   │                Run "pnpm self-update" to update.                 │
   │                                                                  │
   │         Follow @pnpmjs for updates: https://x.com/pnpmjs         │
   │                                                                  │
   ╰──────────────────────────────────────────────────────────────────╯
```

But running `pnpm self-update` gives error:
 ERR_PNPM_CANT_SELF_UPDATE_IN_COREPACK  You should update pnpm with corepack



(I have checked that the bug still exists in the latest version)



### Expected Behavior

Correct instructions given:
corepack install -g pnpm@${opts.latestVersion}



### Which Node.js version are you using?

20.9.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

_No response_

--- a/cli/default-reporter/src/reporterForClient/reportUpdateCheck.ts
+++ b/cli/default-reporter/src/reporterForClient/reportUpdateCheck.ts
@@ -51,11 +51,11 @@ function renderUpdateMessage (opts: UpdateMessageOptions): string {
 }
 
 function renderUpdateCommand (opts: UpdateMessageOptions): string {
-  if (opts.env.PNPM_HOME) {
-    return 'pnpm self-update'
-  }
   if (isExecutedByCorepack(opts.env)) {
     return `corepack install -g pnpm@${opts.latestVersion}`
+  }
+  if (opts.env.PNPM_HOME) {
+    return 'pnpm self-update'
   }
   const pkgName = opts.currentPkgIsExecutable ? '@pnpm/exe' : 'pnpm'
   return `pnpm add -g ${pkgName}`

